android/src/main/io/grpc/android/AndroidChannelBuilder.java

```diff
@@ -0,0 +1,108 @@
+
+/**
+ * A {@link io.grpc.ChannelBuilder} that provides a stable interface for producing
+ * {@link io.grpc.Channel} instances on Android. This allows applications to be resilient
+ * to changes such as defaulting to a new channel implementation or breaking API changes
+ * in existing ones. Configuration options will be added here only if they provide value
+ * to a broad set of users and are likely to be resilient to changes in the underlying channel
+ * implementation.
+ *
+ * <p>Developers needing advanced configuration are free to use the underlying channel
+ * implementations directly while assuming the risks associated with using an
+ * {@link io.grpc.ExperimentalApi}.
+ */
+
+// TODO(lryan):
+// - Document the various dangers of scheduling work onto Android Main/UI/whatever
+// threads. Is any enforcement practical? Conversely is there a sensible default. I assume
+// its also a bad idea to use huge numbers of threads too.
+// - Provide SSL installation and detect ALPN/NPN support?
+// - Allow for an alternate TrustManager ? for self-signed, pinning etc. Is there a good default
+//   impl for these variations blessed by security team?
+// - Augment with user-agent with app id?
+// - Specify a smaller flow-control window by default? Fewer concurrent streams?
+public class AndroidChannelBuilder extends ChannelBuilder<AndroidChannelBuilder> {
+
+  // Currently we only support OkHttp on Android. This will become dynamic as more transport
+  // implementations become available.
+  private final OkHttpChannelBuilder baseBuilder;
+
+  public static AndroidChannelBuilder forAddress(String host, int port) {
+    return new AndroidChannelBuilder(OkHttpChannelBuilder.forAddress(host, port));
+  }
+
+  private AndroidChannelBuilder(OkHttpChannelBuilder builder) {
+    this.baseBuilder = builder;
+  }
+
+  @Override
+  public AndroidChannelBuilder executor(ExecutorService executor) {
+    baseBuilder.executor(executor);
+    return this;
+  }
+
+  public AndroidChannelBuilder intercept(List list) {
+    baseBuilder.intercept(list);
+    return this;
+  }
+
+  @Override
+  public AndroidChannelBuilder intercept(ClientInterceptor... interceptors) {
+    baseBuilder.intercept(interceptors);
+    return this;
+  }
+
+  @Override
+  public AndroidChannelBuilder userAgent(String userAgent) {
+    baseBuilder.userAgent(userAgent);
+    return this;
+  }
+
+  /**
+   * Use of a plaintext connection to the server. By default a secure connection mechanism
+   * such as TLS will be used.
+   *
+   * <p>Should only be used for testing or for APIs where the use of such API or the data
+   * exchanged is not sensitive.
+   *
+   * @param skipNegotiation @{code true} if there is a priori knowledge that the endpoint supports
+   *                        plaintext, {@code false} if plaintext use must be negotiated.
+   */
+  @ExperimentalApi("primarily for testing")
+  public AndroidChannelBuilder usePlaintext(boolean skipNegotiation) {
```

Comment:
So there are three types in io.grpc.netty.NegotiationType: `TLS`, `PLAINTEXT_UPGRADE` and `PLAINTEXT`, why don't we move it to io.grpc package for all transports use? Is there any transport-specified negotiation type expected in the future?